java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -385,11 +384,11 @@ public PubSubMessage tryGetPubSubMessage() {
                     "The operation will never complete since messages will be passed to the configured"
                             + " callback.");
         }
-        return messageQueue.poll();
+        return messageQueue.popSync();
     }
 
     /**
-     * Returns a promise for a next pubsub message.
+     * Async returns a promise for a next pubsub message.
      *
      * @apiNote <b>Not implemented!</b>
      * @throws ConfigurationError If client is not subscribed to any channel or if client configured
```

Comment:
Can you change the text of the @return "A <code>CompletableFuture</code> which will asynchronously hold the next available message.